packages/cli-kit/src/node/base-command.ts

```diff
@@ -27,13 +34,29 @@ abstract class BaseCommand extends Command {
     options?: Interfaces.Input<TFlags> | undefined,
     argv?: string[] | undefined,
   ): Promise<Interfaces.ParserOutput<TFlags, TArgs>> {
-    const result = await super.parse<TFlags, TArgs>(options, argv)
+    const rawResult = await super.parse<TFlags, TArgs>(options, argv)
+    const flags = rawResult.flags as PresettableFlags
+    const presetsDirectory = await this.presetsPath(flags)
+    const findUp = this.findUpForPresets()
+    const result = await super.parse<TFlags, TArgs>(
+      await withPresets({options, rawResult, presetsDirectory, findUp}),
+      argv,
+    )
+    if (flags.preset) {
+      reportDifferences(rawResult.flags, result.flags, flags.preset)
+    }
     await addFromParsedFlags(result.flags)
     return result
   }
-}
 
-export default BaseCommand
+  protected async presetsPath(_rawFlags: {path?: string}): Promise<string> {
```

Comment:
Somewhat related and mentioned when we caught up. It probably makes sense to allow the overriding of the presets file path in an environment variable -- if provided, skip all the search logic and just work from that file.

Reply:
I'm OK with `path` being cemented as a CLI convention. It seems like an eminently reasonable one. Also OK with recursing by default, so essentially themes will have the full naive behavior while apps will be a bit more opinionated.

In terms of `loadPresetsFromDirectory`, I think that it's so specialized that it shouldn't be overridable. If you're not using it, you're probably not really doing presets. And I suspect we might build in more things later, like validations etc. Ultimately the method just gets a directory where your presets file is (potentially) located and converts it to a collection of presets. If we elect to have a more complex file structure, the parsing will be done in that function as well.

Agreed it could be nice to have a flag to specify where your presets file is. However, I don't think it's worth investing more in that direction until the basics are more fleshed out.